### PR TITLE
Fix for missing optional parameters state in getLocals

### PR DESCRIPTION
**Pull request by hyp3rflow**
The `getLocals` function can be used even if some optional parameters are not provided. However, if some optional arguments are missing, the `params zip args` does not consider the value of unprovided optional parameter, where an `AbsentTop` value is expected.

This PR fixes this by ensuring `AbsentTop` is provided for value of missing optional arguments.

### Reproduction

If you want to get quick reproduction, run following REPL command:
```
esmeta tycheck -tycheck:repl -tycheck:target=CreateArrayFromList
```
The `ArrayCreate` function is called at the first instruction. This function requires a 'length' as its first parameter and optionally a 'proto' as the second parameter. However, when `CreateArrayFromList` calls `ArrayCreate`, it doesn't provide this second argument. Consequently, at the entry point, `ArrayCreate` is unable to find 'proto' in its state due to the fact that `params zip args` doesn't provide any value for missing arguments.

This absence of 'proto' leads to a condition `(= proto absent)` in `ArrayCreate` resulting in 'Bot'. This 'Bot' result in turn prevents the addition of any subsequent block (i.e., 'then' or 'else' blocks) into worklist. As a result, the analysis is not fully completed and terminates prematurely.

### CODE REVIEW

File: src/main/scala/esmeta/analyzer/TypeSemantics.scala [hunk 1]
```diff
@@ -71,22 +71,25 @@ class TypeSemantics(
...
-            argsForMismatch += key -> {
-              argsForMismatch.get(key).fold(argTy)(_ || argTy)
-            }
-          AbsValue(paramTy)
-      // force to set expected type for parameters
-      param.lhs -> expected
-    }).toMap
+    (for (((param, arg), idx) <- (params zip argsWithOptional).zipWithIndex)
+      yield {
+        val argTy = removeAbsentTy(arg.ty)
```
Comment (hyp3rflow) [inline src/main/scala/esmeta/analyzer/TypeSemantics.scala:79]: IIRC we previously decided to remove the use of `removeAbsentTy` here, but I've discovered this use is necessary for handling optional parameter correctly.
In the case of optional parameters, their type is their own specified type, excluding `Absent`, even when they are optional. There is another part of our code where we handle the Absent state for optional parameters, so I've decided to not to remove `removeAbsentTy`.
https://github.com/es-meta/esmeta/blob/af37f8a694d749bb9a624de71407eed2fa71b1dc/src/main/scala/esmeta/analyzer/TypeAnalyzer.scala#L90

**Comment by jhnaldo** [comment]
@hyp3rflow Since you are an internal member of ESMeta, I suggest creating a branch in this repository for a PR to check the change's impact. And the merge target of PRs should be the `dev` branch instead of the `main` branch.

**Comment by hyp3rflow** [comment]
@jhnaldo I attempted to push a commit to this repository but encountered a permission error message. Could you please check the repository's permission settings?

**Comment by jhnaldo** [comment]
@hyp3rflow I invited you as a member of ESMeta. Can you check your e-mail about this invitation?

**Comment by hyp3rflow** [comment]
@jhnaldo I did accept the invitation, but I'm still receiving the same error message.

**Comment by jhnaldo** [comment]
@hyp3rflow I'm sorry. I just remembered that I had to give you permission for this repo. I hope it works now.

**Comment by hyp3rflow** [comment]
@jhnaldo thanks, it works! Should I open a new pull request from the branch in this repository?

**Comment by jhnaldo** [comment]
@hyp3rflow Creating a new PR from the branch in this repo is better for me.

**Comment by hyp3rflow** [comment]
Transfers to #151

**Pull request closed without merging**